lib/m3u8-parser.js

```diff
@@ -22,6 +27,26 @@ var m3u8parser = module.exports = function() {
       self.emit('item', line.trim());
     }
   }
+  function ytdlParse() {
+    var time = self.ytdlSavedLines[self.ytdlSavedLines.length - 2];
+    if (!time) {
+      self.emit('ytdl-error', new Error('Time is null/undefined.\n' + time));
+      return;
+    }
+    time = parseInt(time.replace(/#EXTINF:/g, "").replace(/,/g, ""));
+    if (self.isFirstTime) time = 1;
+    var line = self.ytdlSavedLines[self.ytdlSavedLines.length-1];
```

Comment:
Get's the last line in the list. This is the most important part. After every packet that is sent in from the data, the last one in the array is the "live" value. The top one in the array (ignoring all the tags that are at the top) are actually the archive (hence why it started at -4hours when streaming originally). If we want to have future options to start at a certain time, this would be the place to code it.

Reply:
I'd like to just state that right now, I'm realizing we could optionally add 5 second time frames that we can go back in time in. The lower this number, the more 5 seconds we go back in time. So instead of if -1 for the last item (live), we could do -24 for 2 minutes back in the video, and so forth. The limit would have to be -14400, though, as that's four hours back in time (and cannot be guaranteed).